DSharpPlus/Entities/Channel/DiscordThreadChannel.cs

```diff
@@ -0,0 +1,582 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2021 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DSharpPlus.Exceptions;
+using DSharpPlus.Net.Abstractions;
+using DSharpPlus.Net.Models;
+using DSharpPlus.Net.Serialization;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities
+{
+    /// <summary>
+    /// Represents a discord thread channel.
+    /// </summary>
+    public class DiscordThreadChannel : SnowflakeObject, IEquatable<DiscordThreadChannel>
+    {
+        /// <summary>
+        /// Gets ID of the guild to which this thread belongs.
+        /// </summary>
+        [JsonProperty("guild_id", NullValueHandling = NullValueHandling.Ignore)]
+        public ulong GuildId { get; internal set; }
+
+        /// <summary>
+        /// Gets ID of the owner that started this thread.
+        /// </summary>
+        [JsonProperty("owner_id", NullValueHandling = NullValueHandling.Ignore)]
+        public ulong OwnerId { get; internal set; }
+
+        /// <summary>
+        /// Gets ID of the news or text channel that contains this thread.
+        /// </summary>
+        [JsonProperty("parent_id", NullValueHandling = NullValueHandling.Ignore)]
+        public ulong ParentChannelId { get; internal set; }
```

Comment:
Never expose these IDs. Use concrete types for DiscordGuild, DiscordMember and DiscordChannel instead.

Reply:
Added methods to get concrete DiscordGuild and DiscordChannel. Left the id's public however as DiscordChannel leaves these properties public.